src/controllers/auth/login.js

```diff
@@ -124,9 +126,13 @@ define(["apphost", "appSettings", "dom", "connectionManager", "loading", "layout
             html += "</div>";
             html += "</div>";
             html += "</button>";
-        }
+        });
 
-        context.querySelector("#divUsers").innerHTML = html;
+        const divUsers = context.querySelector("#divUsers");
+        divUsers.innerHTML = html;
+        if (!layoutManager.tv) {
+            divUsers.querySelector(".firstCard").focus();
```

Comment:
For TV, I set focus in `showVisualForm` but `autoFocuser` is TV only. In fact, it finds the first focusable element (first card).

Reply:
Hm, I see the following in showManualForm:

```javascript
if (focusPassword) {
    context.querySelector("#txtManualPassword").focus();
} else {
    context.querySelector("#txtManualName").focus();
}
```

Wouldn't that cause issues with the focusManager when in TV layout?

Reply:
Also wouldn't setting focus in showVisualForm lead to the focus never being on the first user card (where it would make the most sense, imo)

Reply:
> Wouldn't that cause issues with the focusManager when in TV layout?

If you call `focusManager.focus(elem)` you will prevent scroll (required by `scrollManager`). But if element is within viewport, there is no difference. But I would prefer `focusManager.focus(elem)` - this is OK for all layouts.

> Also wouldn't setting focus in showVisualForm lead to the focus never being on the first user card (where it would make the most sense, imo)

I think I got what you mean.
Here `showVisualForm` with my focus will be called before user cards are generated.
https://github.com/jellyfin/jellyfin-web/blob/4b69f85bb9dd43e0aa3548d639ffe0a92f5a2f1a/src/controllers/auth/login.js#L210-L211
I think this only works because `require` is "deferred". tbh, I never have a problem with this, but, to be true, this is not ideal.

Originally, my idea was to place focus code into each page, but many of them will be equal. So I combined common logic into `autoFocuser`. Also, it has restoring of paging buttons. But `autoFocuser` is optional, so I think you are free to just focus on first card for all layouts.

Reply:
@dmitrylyzo I removed the focus changes, they'll be better in their own PR down the line.